src/easier68k/core/opcodes/dc.py

```diff
@@ -0,0 +1,392 @@
+        if len(parts) == 1:  # Use the default size
+            size = 'W'
+        else:
+            size = parts[1].upper()
+
+        length = 0.0
+        quote_delim = None  # If None, we're not in a quote: if it has a value, we know what we're looking for to break the quote
+        current_param = ''
+        is_string = False
+
+        for c in parameters:
+            if c == ',' and not quote_delim:  # End of this parameter (and not a comma in a quote)
+                if current_param:
+                    if is_string:
+                        length += len(current_param) * 0.5
+                        if len(current_param) % 2 != 0:
+                            length += 1
+                    else:
+                        if size == 'B':
+                            length += 0.5
+                        elif size == 'W':
+                            length += 1
+                        elif size == 'L':
+                            length += 2
+
+                is_string = False
+                quote_delim = None
+                current_param = ''
+                continue
+
+            if quote_delim and c == quote_delim:  # Breaking out of the quote
+                quote_delim = None
+                continue
+
+            if c == '\'':
+                if not quote_delim:  # This is the start of a quote
+                    quote_delim = '\''
+                    current_param = ''
+                    is_string = True
+                    continue
+
+            if c == '\"':
+                if not quote_delim:  # This is the start of a quote
+                    quote_delim = '\"'
+                    current_param = ''
+                    is_string = True
+                    continue
+
+            current_param += c
+
+        if current_param:
+            if is_string:
+                length += len(current_param) * 0.5
+            else:
+                if size == 'B':
+                    length += 0.5
+                elif size == 'W':
+                    length += 1
+                elif size == 'L':
+                    length += 2
+
+        length = math.ceil(length)
+        if size == 'L' and length % 4 != 0:
+            length = length + (length % 4)
+
+        return length, issues
+
+    def assemble(self) -> bytearray:
+        """
+        Assembles this opcode into hex to be inserted into memory
+        :return: The hex version of this opcode
+        """
+        tr = ''
+        for val in self.values:
+            tr += '{0:08b}'.format(parse_literal(val))
+
+        return bytearray.fromhex(hex(int(tr, 2))[2:])  # Convert to a bytearray
+
+    def execute(self, simulator: M68K):
+        """
+        Executes this command in a simulator
+        :param simulator: The simulator to execute the command on
+        :return: Nothing
+        """
+        pass
+
+    def __str__(self):
+        return "DC command: Size {}, items: {}".format(self.size, self.values)
```

Comment:
I wonder if we could make opcodes more OOP, since we'll have to be making so many of them.
Instead of defining `__str__` for each opcode, could have a value in the class that specifies what the opcode name is, and a list of its parameters.

Reply:
There's already the `class_name` parameter which we can use and then do some introspection to print it out and/or make a method like `get_str` in Opcode which takes the class name and a list of variables and pretty-prints them kinda like this.

On the note of making them more OOP, I did have the idea of making a couple base classes like "two-parameter opcode", which has two assembly parameters and you just define legal types for both of them and then override execute and assemble, while that base class automates some of the parsing.

Reply:
This will be handled in another PR